rust-runtime/aws-smithy-types/src/timeout.rs

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+//! This module defines types that describe timeouts for the various stages of an HTTP request.
+
+use std::borrow::Cow;
+use std::fmt::Display;
+use std::time::Duration;
+
+/// Configuration for the various kinds of timeouts supported by aws_smithy_client::Client.
+#[derive(Debug, Clone, PartialEq, Default)]
+pub struct TimeoutConfig {
+    connect_timeout: Option<Duration>,
+    tls_negotiation_timeout: Option<Duration>,
+    read_timeout: Option<Duration>,
+    api_call_attempt_timeout: Option<Duration>,
+    api_call_timeout: Option<Duration>,
+}
+
+impl TimeoutConfig {
+    /// Create a new `TimeoutConfig` with no timeouts set
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    /// Create a new [TimeoutConfigBuilder]
+    /// A limit on the amount of time after making an initial connect attempt on a socket to complete the connect-handshake.
+    pub fn connect_timeout(&self) -> Option<Duration> {
+        self.connect_timeout
+    }
+
+    /// A limit on the amount of time a TLS handshake takes from when the `CLIENT HELLO` message is
+    /// sent to the time the client and server have fully negotiated ciphers and exchanged keys.
+    pub fn tls_negotiation_timeout(&self) -> Option<Duration> {
+        self.tls_negotiation_timeout
+    }
+
+    /// A limit on the amount of time an application takes to attempt to read the first byte over an
+    /// established, open connection after write request. A.K.A. the "time to first byte" timeout.
+    pub fn read_timeout(&self) -> Option<Duration> {
+        self.read_timeout
+    }
+
+    /// A limit on the amount of time it takes for the first byte to be sent over an established,
+    /// open connection and when the last byte is received from the service for a single attempt.
+    /// If you want to set a timeout for an entire request including retry attempts,
+    /// use [TimeoutConfig::api_call_timeout] instead.
+    pub fn api_call_attempt_timeout(&self) -> Option<Duration> {
+        self.api_call_attempt_timeout
+    }
+
+    /// A limit on the amount of time it takes for request to complete. A single request may be
+    /// comprised of several attemps depending on an app's [super::retry::RetryConfig]. If you want
+    /// to control timeouts for a single attempt, use [TimeoutConfig::api_call_attempt_timeout].
+    pub fn api_call_timeout(&self) -> Option<Duration> {
+        self.api_call_timeout
+    }
+
+    /// Consume a [TimeoutConfig] to creat a new one, setting the connect timeout
+    pub fn with_connect_timeout(mut self, timeout: Duration) -> Self {
+        self.connect_timeout = Some(timeout);
+        self
+    }
+
+    /// Consume a [TimeoutConfig] to creat a new one, setting the TLS negotiation timeout
```

Comment:
Create is typo'd in several of these as "creat"... Maybe do a find/replace.

Reply:
I was clearly referring to the [East Indian herb](https://www.merriam-webster.com/dictionary/creat) here /s